newrelic/resource_newrelic_workload.go

```diff
@@ -103,30 +255,31 @@ func resourceNewRelicWorkloadCreate(ctx context.Context, d *schema.ResourceData,
 		ID:        created.ID,
 		GUID:      created.GUID,
 	}
-
 	d.SetId(ids.String())
+
 	return resourceNewRelicWorkloadRead(ctx, d, meta)
 }
 
 func resourceNewRelicWorkloadRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*ProviderConfig).NewClient
-
 	ids, err := parseWorkloadIDs(d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}
-
-	workload, err := client.Workloads.GetWorkloadWithContext(ctx, ids.AccountID, ids.GUID)
+	workload, err := client.Workloads.GetWorkloadWithContext(ctx, ids.AccountID, string(ids.GUID))
 	if err != nil {
-		if _, ok := err.(*errors.NotFound); ok {
-			d.SetId("")
-			return nil
-		}
+		if err != nil {
+			if _, ok := err.(*errors.NotFound); ok {
```

Comment:
Let's validate if this works. Delete the workload from the UI should trigger this.